fenix/app/src/main/java/org/mozilla/fenix/shopping/middleware/ReviewQualityCheckNavigationMiddleware.kt

```diff
@@ -39,13 +47,27 @@ class ReviewQualityCheckNavigationMiddleware(
     private fun processAction(
         action: ReviewQualityCheckAction.NavigationMiddlewareAction,
     ) = scope.launch {
-        when (action) {
-            is ReviewQualityCheckAction.OpenLink -> {
-                when (action.link) {
-                    is ReviewQualityCheckState.LinkType.ExternalLink -> openLink(action.link.url, true)
-                    is ReviewQualityCheckState.LinkType.AnalyzeLink -> openLink(action.link.url, false)
-                }
-            }
-        }
+        selectOrAddUseCase.invoke(
+            url = actionToUrl(
+                action = action,
+            ),
+        )
+    }
+
+    /**
+     * Used to find the corresponding url to the open link action.
+     *
+     * @param action Used to find the corresponding url.
+     */
+    @VisibleForTesting
+    internal fun actionToUrl(
```

Comment:
I think we can remove  @VisibleForTesting and assert on the browser state, we are asserting for the tab size, we can also assert for the tab url based on the action, and we can make this a private function. 

This way we don't test the implementation detail but the output. So in the future if this class is refactored without any behavior change, tests won't need to be updated and will help in refactoring confidently.